core/src/main/java/com/linecorp/armeria/common/util/BlockingTaskExecutorBuilder.java

```diff
@@ -136,6 +145,7 @@ public BlockingTaskExecutor build() {
         final ScheduledThreadPoolExecutor scheduledThreadPoolExecutor = new ScheduledThreadPoolExecutor(
                 numThreads, threadFactory);
         scheduledThreadPoolExecutor.setKeepAliveTime(keepAliveTimeMillis, TimeUnit.MILLISECONDS);
+        scheduledThreadPoolExecutor.allowCoreThreadTimeOut(allowThreadTimeOut);
```

Comment:
Could you add a test that verifies an `allowThreadTimeOut` is correctly specified?
You can unwrap and access the underlying `scheduledThreadPoolExecutor` by implementing [Unwrappable](https://javadoc.io/doc/com.linecorp.armeria/armeria-javadoc/latest/com/linecorp/armeria/common/util/Unwrappable.html).

Reply:
The unwrapped `ScheduledThreadPoolExecutor` might also be used to solve #3710. /cc @ghkim3221

Reply:
That's a good suggestion. 👍

Reply:
We tried to implement `Unwrappable` only to see that it's hard to make it work as `ScheduledThreadExecutor` is not implementing `Unwrappable` 😭..

Would you please give us more specific guide for this? 😅
Maybe we can just make an `unwrap()` method which returns its backing threadpool..?

Reply:
>  Maybe we can just make an unwrap() method which returns its backing threadpool..?

Yes, it is. You can just return `delegate`.

For example: 
```java
interface BlockingTaskExecutor extends ScheduledExecutorService, Unwrappable {

   
    @Override
    default ScheduledExecutorService unwrap() {
        return this;
    }
}

class DefaultBlockingTaskExecutor implements BlockingTaskExecutor {

    private final ScheduledExecutorService delegate;

    @Override
    public ScheduledExecutorService unwrap() {
        return delegate;
    }
}
```

Reply:
The code seems not to compile. 😅

Reply:
Sorry for the wrong direction.
How about adding `unwrap()` method without implementing `Unwrappable`?
```
interface BlockingTaskExecutor extends ScheduledExecutorService {  
  
    default ScheduledExecutorService unwrap() {
        return this;
    }
}

class DefaultBlockingTaskExecutor implements BlockingTaskExecutor {
    private final ScheduledExecutorService delegate;

    @Override
    public ScheduledExecutorService unwrap() {
        return delegate;
    }
}
```

Reply:
Another quick fix is to change `BlockingTaskExecutor` as a util class and make `BlockingTaskExecutor#build` just return `ScheduledThreadPoolExecutor`?

Reply:
Unfortunately, it is impossible in Armeria 1.x.
It is not annotated with `@UnstableApi`, so we don't change the return type at the moment.

Reply:
> How about adding unwrap() method without implementing Unwrappable?

Reply:
> How about adding unwrap() method without implementing Unwrappable?

+1